validator/src/commands/contact_info/mod.rs

```diff
@@ -18,16 +31,17 @@ pub fn command(_default_args: &DefaultArgs) -> App<'_, '_> {
 }
 
 pub fn execute(matches: &ArgMatches, ledger_path: &Path) {
-    let output_mode = matches.value_of("output");
+    let contact_info_arg = ContactInfoArg::from_clap_arg_match(matches);
+
     let admin_client = admin_rpc_service::connect(ledger_path);
     let contact_info = admin_rpc_service::runtime()
         .block_on(async move { admin_client.await?.contact_info().await })
         .unwrap_or_else(|err| {
             eprintln!("Contact info query failed: {err}");
             exit(1);
         });
-    if let Some(mode) = output_mode {
-        match mode {
+    if let Some(mode) = contact_info_arg.output {
+        match mode.as_str() {
```

Comment:
While technically not part of the intent of this PR, we have an `OutputFormat` type in `cli-output/src/cli_output.rs`. I think we should switch to use that enum, which will avoid the repeated `json` and `json-compact` strings below as well as in the new unit test you're adding.

To confirm, I think `ContactInfoArgs` should contain a single field:
```
ContactInfoArgs {
    pub output: OutputFormat
}
```

Maybe we make that change first and then rebase + land this one on top of it ?